File: www.admin.com/core-api/src/main/java/com/base/coreapi/model/calendar/Slot.java
package com.base.coreapi.model.calendar;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.util.Date;

@Data
@Entity
@NoArgsConstructor
public class Slot {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private long id;

    private Boolean active;

    private Date start;

    @JsonIgnore
    @ManyToOne
    private OrderDay orderDay;

    @OneToOne
    private Reservation reservation;
}

File: www.admin.com/frontend-app/src/app/model/slot.ts
import {Reservation} from "./backend/calendar/reservation";

export class Slot {

  public start: Date;

  public relatedReservation: Reservation;

  public isVaccinateTime: boolean;

  constructor(start: Date, isVaccinateTime: boolean) {
    this.relatedReservation = null;
    this.start = start;
    this.isVaccinateTime = isVaccinateTime;
  }
}
